projects/igniteui-angular/src/lib/grids/state.directive.ts

```diff
@@ -64,23 +77,31 @@ export interface IColumnState {
     searchable: boolean;
 }
 
-const COLUMNS = 'columns';
-const FILTERING = 'filtering';
-const ADVANCED_FILTERING = 'advancedFiltering';
-const SORTING = 'sorting';
-const GROUPBY = 'groupBy';
-const PAGING = 'paging';
-const ROW_SELECTION = 'rowSelection';
-const ROW_PINNING = 'rowPinning';
-const PINNING_CONFIG = 'pinningConfig';
-const CELL_SELECTION = 'cellSelection';
-const COLUMN_SELECTION = 'columnSelection';
+export type GridFeatures = keyof IGridStateOptions;
+
+export interface Feature {
+    getFeatureState: (context: IgxGridStateDirective) => IGridState;
+    restoreFeatureState: (context: IgxGridStateDirective, state: IColumnState[] | IPagingState | ISortingExpression[] |
+        IGroupingState | IFilteringExpressionsTree | GridSelectionRange[] | IPinningConfig | any[]) => void;
+}
 
 @Directive({
     selector: '[igxGridState]'
 })
 export class IgxGridStateDirective {
 
+    /**
+     * @hidden @internal
+     */
+    public features: GridFeatures[] = [];
```

Comment:
Is that supposed to be exposed to users or internal? Because this is a directive, thus there's no template binding that requires `public` props and I don't currently see this used outside the directive atm. Same applies for the `state` and `currGrid`